asm/dscal_test.go

```diff
@@ -0,0 +1,119 @@
+// Copyright ©2016 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package asm
+
+import (
+	"fmt"
+	"testing"
+)
+
+var dscalTests = []struct {
+	alpha float64
+	x     []float64
+	want  []float64
+}{
+	{
```

Comment:
If you haven't, you should see https://github.com/xianyi/OpenBLAS/issues/624

It's not clear to me what we do or do not want to guarantee with NaN, especially if we imagine users will frequently be using cgo.